config/config_test.go

```diff
@@ -127,6 +135,25 @@ limits:
 	}
 }
 
+func TestInvalidHealthCheckConfig(t *testing.T) {
+	buf := bytes.NewBufferString(`
+proxy:
+  handler: http
+  host: http://proxy.example.com
+  listen: localhost:8080
+health-check:
+  enabled: true
+  port: 8080
```

Comment:
Hmmm... sorry to waffle on this... but what are your thoughts on making this more like the `proxy` interface by using a `listen` variable instead of `port`?